hexchat/aescrypt.pl

```diff
@@ -0,0 +1,387 @@
+#!/usr/bin/perl -w
+
+# aescrypt.pl 0.2.2 by OJ Reeves <oj@buffered.io> @TheColonial
+# 					Ported by Lockyc
+# Encrypts chats using AES. Inspired by the blowjob.pl script.
+# Perl modules required:
+#     - Crypt::CBC
+#     - Crypt::Rijndael
+#     - JSON
+#     - MIME::Base64
+#     - Digest::SHA
+#     - Try::Tiny
+# Features:
+#     - Per user/channel AES encryption keys and IV
+#     - Payloads sent as JSON, with checksum for payload validation
+#     - Single messages via /aes <message>
+#     - Enable for all messages via /aeson
+#     - Disable for all messages via /aesoff
+#     - Show configuration via /aesshow
+#     - Doesn't screw with bells and other irssi features like the
+#       blowjob.pl script does
+
+use strict;
+#use warnings;
+
+use Crypt::CBC;
+use JSON;
+use Try::Tiny;
+use MIME::Base64;
+use Digest::SHA qw(sha256_base64);
+
+use HexChat qw(:all);
+
+use vars qw($VERSION %IRSSI $cipher);
+
+$VERSION = "0.1.0";
+%IRSSI = (
+	authors => 'OJ Reeves',
+	contact => 'oj@buffered.io',
+	name => 'aescrypt',
+	description => 'This script encrypts IRC communications using AES in CBC mode. The script supports per-user and per-channel keys and IVs which can be set via irssi commands. Encrypted content is compresed and Base64 encoded prior to being transmitted.',
+	license => 'Apache 2.0',
+	url => 'https://github.com/OJ/aescript'
+);
+HexChat::register($IRSSI{name}, $VERSION, $IRSSI{description});
```

Comment:
I'd be happy for you to rip out all the IRSSI related stuff completely and just have HexChat related stuff instead mate. These headers also give me too much credit for your own stuff. They should reflect your effort, not mine, and just say "based on.." :)